### PR TITLE
connect: remove bowser dependency

### LINKED ISSUE #5324: @trezor/connect-popup: share browser detection code with suite
- [ ] remove `connect/src/utils/browserUtils`
- [ ] implement shared browser detection in @trezor/connect-popup and @trezor/connect-iframe

### PR DESCRIPTION
**Pull request by mroz22**
ok, stay Awhile and listen. 

in #6509 I would like to use [scheduleAction util](https://github.com/trezor/trezor-suite/blob/develop/packages/utils/src/scheduleAction.ts#L102) which internally uses `AbortController` which does not work very well in our tests, for example [coinjoin package](https://github.com/trezor/trezor-suite/blob/2f3f45da3ac54e17d63c9465683d813214f22df7/packages/coinjoin/jest.config.js) needs to use `jsdom` testEnvironment where it miraculously works, but it is actually not right, we want to test such package in `node` env. 

My problem is similar, after doing this change in transport, all the tests in connect start to [fail](https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/3932284548#L3351) and I cannot really do any workarounds here. So what are my options? I could somehow tweak our jest configs, but since we are still on version 25 or which one and the latest one is 29 it feels like a waste of time to study outdated docs. So it only makes sense to update jest and see if it does not fix the problem (it does).

It is not the best situation we can have - having 2 different versions of jest in dependencies but I believe this is the only way how to update it, package per package. Last time I tried to do it globally I gave up after few horus because it comes with many problems. cc @tomasklim 

You may also notice that I have created a new package called `@trezor/connect-browser-utils`. There are still couple of references to browser specifics in `@trezor/connect` and it shouldn't be like this. That package should be pure node.js. Adding this package and doing some refactoring I was able to completly get rid of `bowser` dependency from `@trezor/connect` which is good I believe. Originally, my motivation was to be able to switch testEnvironment for @trezor/connect [__unit__ tests](https://github.com/trezor/trezor-suite/blob/develop/packages/connect/jest.config.js#L6) to 'node' but I did not succeed doing this. There are still some window references elswhere...  
  
related #5319
resolve #5324
related #6025

### CODE REVIEW

**Comment by mroz22** [comment]
### update 

based on discussion with @szymonlesisz and also on commit of his [f28d327](https://github.com/trezor/trezor-suite/pull/7844/commits/f28d3271736f7ef3037f11158a59693388fbbbd6) which I shamelessly took and squashed we can achieve basically the same but we dont need to use  `@trezor/connect-browser-utils` check commits, I hope they are self-explanatory :D

File: packages/env-utils/src/envUtils.native.ts [hunk 1]
```diff
@@ -8,11 +8,13 @@ const getUserAgent = () => '';
...
 
+const getBrowserName = () => '';
+
 const getBrowserVersion = () => '';
 
-const getOsVersion = () => `${Platform.Version}`;
+const getDeviceType = () => '';
 
-const getBrowserName = () => '';
+const getOsVersion = () => `${Platform.Version}`;
```
Comment (mroz22) [inline packages/env-utils/src/envUtils.native.ts:17]: just reordering some of the lines ~alphabetically

File: packages/connect-popup/src/view/selectDevice.ts [hunk 2]
```diff
@@ -123,12 +122,16 @@ export const selectDevice = (payload: UiRequestSelectDevice['payload']) => {
...
-                const os = getOS();
+                const { systemInfo } = getState();
                 // default explanation: contact support
 
                 let explanationContent = `Please <a href="${TREZOR_SUPPORT_URL}" target="_blank" rel="noreferrer noopener" onclick="window.closeWindow();">contact support.</a>`;
                 // linux + LIBUSB_ERROR handling
-                if (os === 'linux' && device.error.indexOf(ERRORS.LIBUSB_ERROR_MESSAGE) >= 0) {
+                if (
+                    systemInfo?.os?.name &&
+                    systemInfo.os.name.toLocaleLowerCase() === 'linux' &&
```
Comment (mroz22) [inline packages/connect-popup/src/view/selectDevice.ts]: this does not play very well. there could be Ubuntu, Mint, Linux...

Comment (mroz22) [reply packages/connect-popup/src/view/selectDevice.ts]: maybe like this? 830a81d5615e10d7694939925a66dcf542905583 fae721e856a2dcb0863fd47e45ff5ffc28ada46c
or maybe I could use ['rpm32','rpm64', 'deb32', 'deb64'].includes(systemInfo.platform)

File: packages/connect-iframe/src/systemInfo.ts [hunk 3]
```diff
@@ -44,72 +42,27 @@ const getSuggestedPlatform = (): Platform | undefined => {
...
-        version: getOsName(),
-        name: getOsVersion(),
-    };
-
+export const getSystemInfo = (supportedBrowsers: {
+    [key: string]: { version: number };
+}): SystemInfo => {
+    const browserName = getBrowserName();
+    const browserVersion = getBrowserVersion();
+    const supportedBrowser = browserName ? supportedBrowsers[browserName.toLowerCase()] : undefined;
```
Comment (mroz22) [inline packages/connect-iframe/src/systemInfo.ts:40]: these are the possible browser names:

```
2345Explorer, 360 Browser, Amaya, Android Browser, Arora, Avant, Avast, AVG,
BIDUBrowser, Baidu, Basilisk, Blazer, Bolt, Brave, Bowser, Camino, Chimera,
Chrome Headless, Chrome WebView, Chrome, Chromium, Cobalt, Comodo Dragon, Dillo,
Dolphin, Doris, DuckDuckGo, Edge, Electron, Epiphany, Facebook, Falkon, Fennec, 
Firebird, Firefox [Focus/Reality], Flock, Flow, GSA, GoBrowser, Huawei Browser, 
ICE Browser, IE, IEMobile, IceApe, IceCat, IceDragon, Iceweasel, Instagram, 
Iridium, Iron, Jasmine, Kakao[Story/Talk], K-Meleon, Kindle, Klar, Konqueror, 
LBBROWSER, Line, LinkedIn, Links, Lunascape, Lynx, MIUI Browser, Maemo Browser, 
Maemo, Maxthon, MetaSr Midori, Minimo, Mobile Safari, Mosaic, Mozilla, NetFront, 
NetSurf, Netfront, Netscape, NokiaBrowser, Obigo, Oculus Browser, OmniWeb, 
Opera Coast, Opera [Mini/Mobi/Tablet], PaleMoon, PhantomJS, Phoenix, Polaris, 
Puffin, QQ, QQBrowser, QQBrowserLite, Quark, QupZilla, RockMelt, Safari, 
Sailfish Browser, Samsung Browser, SeaMonkey, Silk, Skyfire, Sleipnir, Slim, 
SlimBrowser, Swiftfox, Tesla, Tizen Browser, UCBrowser, UP.Browser, Viera, 
Vivaldi, Waterfox, WeChat, Weibo, Yandex, baidu, iCab, w3m, Whale Browser...
```
for example it has `Firefox [Focus/Reality]`not sure how it works in reality.

**Review by szymonlesisz — APPROVED** [review]

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**